include/zephyr/net/coap.h

```diff
@@ -696,6 +696,27 @@ static inline uint16_t coap_block_size_to_bytes(
 	return (1 << (block_size + 4));
 }
 
+/**
+ * @brief Helper for converting block size in bytes to enumeration.
+ *
+ * NOTE: Only valid CoAP block sizes map correctly.
+ *
+ * @param bytes CoAP block size in bytes.
+ * @return enum coap_block_size
+ */
+static inline enum coap_block_size coap_bytes_to_block_size(uint16_t bytes)
+{
+	int sz = __builtin_ctz(bytes) - 4;
```

Comment:
Shouldn't we use `u32_count_trailing_zeros()` here as I think the `__builtin_ctz` is a gcc specific, and that inlined version uses the builtin variant if found.

Reply:
Yes, I think so. That is a good suggestion.

I was only aware of `stdc_trailing_zeros()` which comes in C23, so GCC did not seem to support it yet.

Let me refactor...